src/middleware/TestingHandler.ts

```diff
@@ -0,0 +1,265 @@
+	 * @param {string} requestID - request id
+	 * @param {Date} errDate - date of the request
+	 * @returns response body
+	 */
+	private createResponseBody(statusCode: number, statusMessage: string, requestID: string, requestDate: Date) {
+		// response body gets created as empty for passed cases
+		// response body contains error field for failure scenarios
+		let responseBody;
+		if (statusCode >= 400) {
+			const codeMessage: string = httpStatusCode[statusCode];
+			const errMessage: string = statusMessage;
+
+			responseBody = {
+				error: {
+					code: codeMessage,
+					message: errMessage,
+					innerError: {
+						"request-id": requestID,
+						date: requestDate,
+					},
+				},
+			};
+		} else {
+			responseBody = {};
+		}
+
+		return responseBody;
+	}
+
+	/**
+	 * @private
+	 * creates a response object out of responseHeader and responseBody
+	 * @param {TestingHandlerOptions} testingHandlerOptions - The TestingHandlerOptions object
+	 * @param {string} requestURL - the URL for the request
+	 * @returns Response object
+	 */
+	private createResponse(testingHandlerOptions: TestingHandlerOptions, requestURL: string, requestMethod: string): Response {
+		try {
+			// creates a response Object out of responseHeader and responseBody
+			let responseBody;
+			let responseHeader;
+			let requestID: string;
+			let requestDate: Date;
+
+			this.setStatusCode(testingHandlerOptions, requestURL, requestMethod);
+			requestID = generateUUID();
+			requestDate = new Date();
+			responseHeader = this.createResponseHeaders(testingHandlerOptions.statusCode, testingHandlerOptions.statusMessage, requestID, requestDate);
+			responseBody = this.createResponseBody(testingHandlerOptions.statusCode, testingHandlerOptions.statusMessage, requestID, requestDate);
+			const init = { url: requestURL, status: testingHandlerOptions.statusCode, statusText: testingHandlerOptions.statusMessage, headers: responseHeader };
+			const response = new Response(responseBody, init);
+
+			return response;
+		} catch (error) {
+			throw error;
+		}
+	}
+
+	/**
+	 * @private
+	 * Fetches a random status code for the RANDOM mode from the predefined array
+	 * @param {string} requestMethod - the API method for the request
+	 * @returns the random status code
+	 */
+	private getRandomStatusCode(requestMethod: string): number {
+		try {
+			// returns random status code for the random method from the array present
+			const statusCodeArray: number[] = methodStatusCode[requestMethod] as number[];
+			return statusCodeArray[Math.floor(Math.random() * statusCodeArray.length)];
+		} catch (error) {
+			throw error;
+		}
+	}
+
+	/**
+	 * @private
+	 * To fetch the relative URL out of the complete URL
+	 * @param {RegExp} pattern - the regex pattern for the URL
+	 * @param {string} urlMethod - the complete URL
+	 * @returns the relative URL
+	 */
+	private getRelativeURL(pattern: RegExp, urlMethod: string): string {
```

Comment:
If the purpose of the method is to return relative url from the absolute url then it makes sense to have the regex pattern inside the function.

urlMethod needs to be changed to url.